backend/consensus/base.py

```diff
@@ -76,7 +110,9 @@ async def _run_consensus(self):
                                     TransactionsProcessor(session),
                                     ChainSnapshot(session),
                                     AccountsManager(session),
-                                    session,
+                                    lambda contract_address, session=session: ContractSnapshot(
```

Comment:
what is this magic? ^^

Reply:
This is a factory function to get contract snapshots. This is used for dependency injection of the ContractSnapshot, allowing us to create unit tests for it

I wanted to go with just
```py
lambda contract_address: ContractSnapshot(
    contract_address, session
),
```

But Pylance was telling me that
`Add a parameter to the parent lambda function and use variable "session" as its default value; The value of "session" might change at the next loop iteration. [+2 locations]sonarlint(python:S1515)`